tests/main/lxd-mount-units/task.yaml

```diff
@@ -0,0 +1,50 @@
+summary: Test mount units generated for snaps inside lxd container are correct.
+
+details: |
+    Test that mount units generated by snapd-generator inside lxd container
+    are correct for snapfuse.
+
+# only 20.10+, we want lxd images that come with snaps preinstalled.
+systems: [ubuntu-20.10*]
+
+execute: |
+    echo "Install lxd"
+    snap install lxd
+
+    lxd waitready
+    lxd init --auto
+
+    VERSION_ID="$(. /etc/os-release && echo "$VERSION_ID" )"
+    lxd.lxc launch --quiet "ubuntu:$VERSION_ID" ubuntu
+
+    lxd.lxc file push --quiet "$GOHOME"/snapd_*.deb "ubuntu/root/"
+
+    DEB=$(basename "$GOHOME"/snapd_*.deb)
+    lxd.lxc exec ubuntu -- apt install -y /root/"$DEB"
+    lxd.lxc restart ubuntu
+
+    echo "Sanity check that mount overrides were generated inside the container"
+    lxd.lxc exec ubuntu -- find /var/run/systemd/generator/ -name container.conf | MATCH "/var/run/systemd/generator/snap-core18.*mount.d/container.conf"
+    lxd.lxc exec ubuntu -- test -f /var/run/systemd/generator/snap.mount
+
+    echo "Sanity check that core18 snap is mounted correctly"
+    # Make sure core18 is mounted and readable for a regular user
+    retry -n 5 --wait 1 sh -c 'lxd.lxc exec ubuntu -- sudo --user ubuntu --login ls /snap/core18/current'
+    retry -n 5 --wait 1 sh -c 'lxd.lxc exec ubuntu -- sudo --user ubuntu --login mount | grep core18| grep allow_other'
+
+    echo "Check that snapfuse mount options of generated units match those generated by snapd"
+    # the only way to get mount options used by snapd is to install a new snap
+    retry -n 5 --wait 1 sh -c 'lxd.lxc exec ubuntu snap install hello-world'
```

Comment:
now that this test is downloading a snap to install it, this test needs to setup a proxy

Reply:
Right, done.